google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsTracer.java

```diff
@@ -268,6 +273,15 @@ public void grpcMessageSent() {
     grpcMessageSentDelay.set(attemptTimer.elapsed(TimeUnit.NANOSECONDS));
   }
 
+  /*
+  In GrpcClientCalls we set the timeout in ApiContext on CallOptions. The timeout in ApiContext will be the attempt
+  timeout for the first few requests or the remaining operation timeout after retries and back offs.
+  */
+  @Override
+  public void setRemainingDeadline(long deadline) {
+    this.deadline = deadline - operationTimer.elapsed(TimeUnit.MILLISECONDS);
```

Comment:
here we should simply record the deadline, and in `recordAttemptCompletion`, we caluclate and record `deadline - operationTimer.elapsed(TimeUnit.MILLISECONDS);` as the metric.